src/create-extension-pack.js

```diff
@@ -0,0 +1,184 @@
+    }).argv;
+
+const PACKAGE_JSON = 'package.json'
+const categories = ['Extension Packs'];
+const packName = 'builtin-extension-pack';
+const publisher = 'eclipse-theia';
+const repository = 'https://github.com/eclipse-theia/vscode-builtin-extensions';
+
+(async () => {
+    const vscodeVer = await vscodeVersion();
+    const packVersion = await computeVersion(tag);
+    const extPackNameAndVersion = packName + '-' + packVersion;
+
+    const extPackVsixPath = dist(packName + '-' + packVersion + '.vsix');
+    const extensionPackAlreadyAvailable = await isAvailable(extPackVsixPath, packName, packVersion, publisher);
+    if (extensionPackAlreadyAvailable && !force) {
+        console.log("Exiting as this extension package is already created or published: " + extPackVsixPath);
+        return;
+    }
+
+    const extPackSrcFolder = theiaExtension(extPackNameAndVersion);
+    if (!fs.existsSync(extPackSrcFolder)) {
+        await fs.mkdir(extPackSrcFolder);
+    }
+
+    const extPack = {};
+    extPack.name = packName;
+    extPack.displayName = packName;
+    extPack.description = 'Builtin extension pack associated to a version of vscode';
+    extPack.version = packVersion;
+    extPack.publisher = publisher;
+    extPack.categories = categories;
+    extPack.engines = { vscode: '^' + vscodeVer };
+    extPack.repository = repository;
+    extPack.extensionPack = await resolveExtensions();
+
+    if (extPack.extensionPack.length === 0) {
+        process.exitCode = 1;
+        console.error('Aborting: No extension was found available for this version: ' + packVersion);
+        return;
+    }
+
+    const packFolderPath = path.join(extPackSrcFolder, '..', extPackNameAndVersion)
+    const packJsonPath = path.join(packFolderPath, PACKAGE_JSON);
+    const licensePath = path.join(packFolderPath, 'LICENSE.txt');
+    const readmePath = path.join(packFolderPath, 'README.md');
+
+    fs.writeFileSync(packJsonPath, JSON.stringify(extPack, null, 2), 'utf-8');
+    console.log('Generated ' + PACKAGE_JSON + ' file at: ' + packJsonPath);
+    fs.writeFileSync(licensePath, generateLicense());
+    fs.writeFileSync(readmePath, generateReadme());
+
+    await vsce.createVSIX({
+        'cwd': packFolderPath,
+        'packagePath': dist(),
+        'useYarn': true
+    });
+
+    async function resolveExtensions() {
+        const extensionsArr = [];
+        for (const extension of fs.readdirSync(extensions())) {
+            const extDataPath = extensions(extension, PACKAGE_JSON);
+            if (!fs.existsSync(extDataPath)) {
+                console.log('No ' + PACKAGE_JSON + ' found for: ' + extension);
+                continue;
+            }
+
+            const content = fs.readFileSync(extDataPath, 'utf-8');
+            const extData = JSON.parse(content);
+
+            const extVsixPath = dist(extData.name + '-' + packVersion + '.vsix');
+            const available = await isAvailable(extVsixPath, extData.name, packVersion);
```

Comment:
minor: we should rename, or drop the variable entirely.

```suggestion
            const isAvailable = await isAvailable(extVsixPath, extData.name, packVersion);
```

Reply:
the new name would collide with the function name, so I have removed the variable instead.